sys/include/wtimer.h

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_wtimer System Timer
+ * @ingroup   sys
+ * @brief     Provides a high level abstraction timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it. It can deal with low-level timers with less than
+ * 32bit width.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
+ * realized by next-first singly linked lists.
+ *
+ * @{
+ * @file
+ * @author Kaspar Schleiser <kaspar@schleiser.de>
+ */
+#ifndef WTIMER_H
+#define WTIMER_H
+
+#include <stdint.h>
+#include "msg.h"
+#include "periph/timer.h"
+#include "timex.h"
+
+/**
+ * @def internal define to allow using variables instead of defines
+ */
+#ifdef WTIMER_TRACE
+#include "wtimer_trace.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief wtimer callback type
+ */
+typedef void (*timer_callback_t)(void*);
+
+/**
+ * @brief wtimer timer structure
+ */
+typedef struct wtimer {
+    struct wtimer *next;        /**< reference to next timer in timer lists */
+    uint32_t target;            /**< lower 32bit absolute target time */
+    uint32_t long_target;       /**< upper 32bit absolute target time */
+    timer_callback_t callback;  /**< callback function to call when timer expires */
+    void *arg;                  /**< argument to pass to callback function */
+} wtimer_t;
+
+/**
+ * @brief wtimer initialization function
+ *
+ * This sets up wtimer. Has to be called once at system boot.
```

Comment:
How about this: "This must be called before any other wtimer functions are called."
Less risk of misunderstanding IMO.